contracts/account/extensions/ERC7579Modules/ERC7579SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,453 @@
+    /// @param account The ERC-7579 Account to get guardians for
+    /// @return An array of all guardians
+    function getGuardians(address account) public view virtual returns (address[] memory) {
+        return _recoveryConfigs[account].guardians.values();
+    }
+
+    /// @notice Gets the threshold for an ERC-7579 Account
+    /// @param account The ERC-7579 Account to get the threshold for
+    /// @return The threshold value
+    function getThreshold(address account) public view virtual returns (uint256) {
+        return _recoveryConfigs[account].threshold;
+    }
+
+    /// @notice Gets the timelock for an ERC-7579 Account
+    /// @param account The ERC-7579 Account to get the timelock for
+    /// @return The timelock value
+    function getTimelock(address account) public view virtual returns (uint256) {
+        return _recoveryConfigs[account].timelock;
+    }
+
+    /// @notice Gets the maximum number of guardians for an ERC-7579 Account
+    /// @return The maximum number of guardians
+    function maxGuardians() public pure virtual returns (uint256) {
+        return 32;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice EIP-712 digest for starting recovery
+    /// @param account The ERC-7579 Account to start recovery for
+    /// @param executionCalldata The calldata to execute during recovery
+    /// @return The EIP-712 digest for starting recovery
+    function _getStartRecoveryDigest(
+        address account,
+        bytes calldata executionCalldata
+    ) internal view returns (bytes32) {
+        bytes32 structHash = keccak256(
+            abi.encode(START_RECOVERY_TYPEHASH, account, keccak256(executionCalldata), nonces(account))
+        );
+        return _hashTypedDataV4(structHash);
+    }
+
+    /// @notice EIP-712 digest for cancelling recovery
+    /// @param account The ERC-7579 Account to cancel recovery for
+    /// @param nonce The nonce of the account
+    /// @return The EIP-712 digest for cancelling recovery
+    function _getCancelRecoveryDigest(address account, uint256 nonce) internal view returns (bytes32) {
+        bytes32 structHash = keccak256(abi.encode(CANCEL_RECOVERY_TYPEHASH, account, nonce));
+        return _hashTypedDataV4(structHash);
+    }
+
+    /// @notice Verifies multiple guardian signatures for a given digest
+    /// @dev Ensures signatures are unique, and threshold is met
+    /// @param account The account the signatures are for
+    /// @param guardianSignatures Array of guardian signatures sorted by signer address in ascending order
+    /// @param digest The digest to verify the signatures against
+    function _validateGuardianSignatures(
+        address account,
+        GuardianSignature[] calldata guardianSignatures,
+        bytes32 digest
+    ) internal view virtual {
```

Comment:
It'd be better if this function returns a boolean instead of reverting. This way there can be a public view getter called `validateGuardianSignatures` that allows to test if the guardian signatures construction is correct